cluster-autoscaler/core/autoscaler.go

```diff
@@ -66,6 +67,13 @@ func NewAutoscaler(opts AutoscalerOptions) (Autoscaler, errors.AutoscalerError)
 	if err != nil {
 		return nil, errors.ToAutoscalerError(errors.InternalError, err)
 	}
+
+	// These metrics should be published only once.
```

Comment:
This doesn't take into account node groups which get added after Cluster Autoscaler starts.

Reply:
Both the list of nodegroups and min/max size of existing nodegroups can change during cloudprovider.Refresh() call that happens every loop. So even for nodegroups that are already registered at this point the value may change later.

Reply:
This is now addressed by publishing the metrics after each cloud provider refresh in the static autoscaler `RunOnce`